### PR TITLE
[REVIEW] Remove noqa marks from __init__.py files

### PR DESCRIPTION
**Pull request by mrocklin**
These are already handled with the exclusions setup.cfg file

### CODE REVIEW

**Comment by kkraus14** [comment]
Looks like the style checker is complaining about this, I believe the proper pep8 way is to include the `__all__` attribute with the modules.

File: python/cudf/io/__init__.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # Copyright (c) 2018, NVIDIA CORPORATION.
-from .csv import read_csv  # noqa: F401
+from .csv import read_csv
```
Comment (kkraus14) [inline python/cudf/io/__init__.py]:
```suggestion
from .csv import read_csv


__all__ = ['read_csv']

```

**Comment by mrocklin** [comment]
It looks like `flake8` isn't picking up the `setup.cfg` file ?  It might be more of an issue with CI than with the code.

File: python/cudf/dataframe/__init__.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-from cudf.dataframe.buffer import Buffer                # noqa: F401
-from cudf.dataframe.numerical import NumericalColumn    # noqa: F401
-from cudf.dataframe.datetime import DatetimeColumn      # noqa: F401
+from cudf.dataframe.dataframe import DataFrame
+from cudf.dataframe.index import (Index, GenericIndex,
+    RangeIndex, DatetimeIndex)
+from cudf.dataframe.series import Series
+from cudf.dataframe.buffer import Buffer
+from cudf.dataframe.numerical import NumericalColumn
+from cudf.dataframe.datetime import DatetimeColumn
```
Comment (kkraus14) [inline python/cudf/dataframe/__init__.py]:
```suggestion
from cudf.dataframe.datetime import DatetimeColumn


__all__ = ['buffer', 'dataframe', 'series', 'index', 'numerical', 'datetime',
           'DataFrame', 'Index', 'GenericIndex', 'RangeIndex',
           'DatetimeIndex', 'Series', 'Buffer', 'NumericalColumn',
           'DatetimeColumn']

```

Comment (mrocklin) [reply python/cudf/dataframe/__init__.py]: I disagree with these commits.  They're unnecessary and add administrative burden when we add things to the API.

**Comment by mrocklin** [comment]
The style failures are consistent with `flake8` not seeing `setup.cfg`.  Try running `flake8` locally.  Try removing the `__init__.py` entry from the exclusions and running it again and you'll see that that's consistent with what we're seeing in the CI failures.

File: python/cudf/__init__.py [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-from cudf.dataframe import Series      # noqa: F401
-from cudf.multi import concat          # noqa: F401
-from cudf.io import read_csv           # noqa: F401
-from cudf.settings import set_options  # noqa: F401
+from cudf.dataframe import DataFrame
+from cudf.dataframe import Index
+from cudf.dataframe import Series
+from cudf.multi import concat
+from cudf.io import read_csv
+from cudf.settings import set_options
```
Comment (kkraus14) [inline python/cudf/__init__.py]:
```suggestion
from cudf.settings import set_options


__all__ = ['dataframe', 'DataFrame', 'Index', 'Series', 'concat', 'read_csv',
           'set_options']
```

**Comment by kkraus14** [comment]
> The style failures are consistent with `flake8` not seeing `setup.cfg`. Try running `flake8` locally. Try removing the `__init__.py` entry from the exclusions and running it again and you'll see that that's consistent with what we're seeing in the CI failures.

Looks like `setup.cfg` is expecting `__init__.py` in the top level directory instead of the `cudf` directory so it's just a pathing issue I think?

**Comment by mrocklin** [comment]
```
mrocklin@dgx16:~/cudf$ flake8 python/
mrocklin@dgx16:~/cudf$
```

**Comment by kkraus14** [comment]
cc @raydouglass do you know where / how flake8 is being invoked in CI? I believe it's this script: https://github.com/rapidsai/cudf/blob/branch-0.5/ci/checks/style.sh but unsure what the CWD would be for that executing

**Comment by raydouglass** [comment]
> cc @raydouglass do you know where / how flake8 is being invoked in CI? I believe it's this script: https://github.com/rapidsai/cudf/blob/branch-0.5/ci/checks/style.sh but unsure what the CWD would be for that executing

If that script exists, it is run as `source ci/checks/style.sh` from the root of the git repo checkout. If there is no script, `flake8` is invoked in the root.

**Comment by mrocklin** [comment]
I tried taking a look at the script but it appears to be hard-coded to a particular environment and conda environment.  Not sure how to test locally.

In the mean time, I'll try pushing up some small things here and seeing if they work :)

**Comment by mrocklin** [comment]
We don't check `__init__.py` files with the linter (for precisely this situation).  This was already specified in the `setup.cfg` file (which includes exceptions like this).  So we never really should have had the add them in the first place.

However it looks like `flake8` wasn't seeing these exclusions because it was run in the root directory and wasn't pointed to the python directory.  The change that made this pass was changing the `flake8` command to `flake8 python/`, pointing it to the Python directory.  This change is good regardless if we decide that we want to change our flake8 style guidelines in the future in some other way (like extending line lengths).

**Comment by harrism** [comment]
I wish I had known that when I added them. I guess in the refactor the path must have changed and broken this.

**Comment by mrocklin** [comment]
> I wish I had known that when I added them. I guess in the refactor the path must have changed and broken this.

There are too many things to know :)

**Pull request merged**